tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/CallStackDebugTest.java

```diff
@@ -55,6 +57,10 @@ public void setup() throws BallerinaTestException {
     public void callStackTest() throws BallerinaTestException {
         Pair<BallerinaTestDebugPoint, StoppedEventArguments> debugHitInfo = debugTestRunner.waitForDebugHit(25000);
         StackFrame[] frames = debugTestRunner.fetchStackFrames(debugHitInfo.getRight());
+        Thread[] threads = debugTestRunner.fetchThreads();
+
+        // Call stack representation test for single strand visibility.
+        Assert.assertEquals(threads.length, 1);
```

Comment:
IMO it would be better if we can separate out strand related tests from stack trace tests, by moving these strand related testing into a separate test case. In there, we can assert strand representation(count) under different contexts, like
- inside main function
- inside a function which runs asynchronously (invoked using `start` action)
- inside a worker
- inside the init function of a class

Reply:
IMO, strand and stack frame related tests should be together. Tests should be written in such a way that given different scenarios how many strands are running and what are the stack frames belongs to each strand.

With the current thread filtration mechanism, there will be only one main strand throughout the execution flow(main, init and other functions). For workers and `start` action, strands will be created and attached to the currently running threads. We can further improve these test once we support multiple strands debug in future.

Reply:
+1 for moving strand related tests into a separate test case.